### ISSUE TITLE
1.3 beta 7 Binding arguments in strict mode

### ISSUE THREAD

**Issue by rclai** [meteor/meteor#6249]
Attempted to use the `exchanger` NPM package and got this error when trying to do
`import exchanger from 'exchanger';`

```
While building for os.linux.x86_64:
   node_modules/exchanger/node_modules/soap/lib/client.js:79:44: Binding arguments in strict mode (79:44)
   
   While computing assigned variables:
   node_modules/exchanger/node_modules/soap/lib/client.js: Binding arguments in strict mode (79:44)
```

**Comment by rdickert** [comment]
Can you describe exactly how to reproduce this? `npm install exchanger` fails. A small repro app would help. Thanks!

**Comment by rclai** [comment]
Huh that's weird. It worked for me. Let me check on my Npm version.

**Comment by rclai** [comment]
I didn't think a repro was needed.

When I installed it, this is all I did:
```
me@me-dev:~/vuezz# npm --version
1.4.28
me@me-dev:~/vuezz# node --version
v0.10.36
me@me-dev:~/vuezz# npm i --save exchanger
npm WARN package.json @ No description
npm WARN package.json @ No repository field.
npm WARN package.json @ No README data
npm WARN engine exchanger@0.1.1: wanted: {"node":"0.8.x"} (current: {"node":"0.10.36","npm":"1.4.28"})
npm WARN engine boom@2.10.1: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.36","npm":"1.4.28"})
npm WARN engine cryptiles@2.0.5: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.36","npm":"1.4.28"})
npm WARN engine hoek@2.16.3: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.36","npm":"1.4.28"})
-
> node-expat@1.6.0 install /me/vuezz/node_modules/exchanger/node_modules/soap/node_modules/node-expat
> node-gyp rebuild

make: Entering directory `/me/vuezz/node_modules/exchanger/node_modules/soap/node_modules/node-expat/build'
  CXX(target) Release/obj.target/node_expat/node-expat.o
  SOLINK_MODULE(target) Release/obj.target/node_expat.node
  SOLINK_MODULE(target) Release/obj.target/node_expat.node: Finished
  COPY Release/node_expat.node
make: Leaving directory `/me/vuezz/node_modules/exchanger/node_modules/soap/node_modules/node-expat/build'
exchanger@0.1.1 node_modules/exchanger
├── xml2js@0.1.14 (sax@1.1.5)
├── underscore@1.3.3
├── moment@1.6.2
└── soap@0.2.4 (request@2.69.0, node-expat@1.6.0)
```

**Comment by abernix** [comment]
I tried to recreate this problem and it worked fine.  Since `exchanger` requires some module compilation with `node-gyp` (which will vary between node versions) make sure you always use `meteor npm install` wherever you would normally use `npm install` to ensure the compilation Node versions will matching the Meteor runtime node version.

Closing, but feel free to ask for a reopen if you have more details.

(p.s. I also had to `meteor npm install --save compress` as instructed by the meteor tool because the `soap` dependency of `exchanger` doesn't properly specify it's dependencies).
